pex/tools/commands/venv.py

```diff
@@ -61,12 +63,206 @@ def _copytree(
+        if pex_script:
+            script_path = os.path.join(bin_dir, pex_script)
+            os.execv(script_path, [script_path] + sys.argv[1:])
+
+        pex_interpreter = pex_overrides.get("PEX_INTERPRETER", "").lower() in ("1", "true")
+        PEX_INTERPRETER_ENTRYPOINT = "code:interact"
+        entry_point = (
+            PEX_INTERPRETER_ENTRYPOINT
+            if pex_interpreter
+            else pex_overrides.get("PEX_MODULE", {entry_point!r} or PEX_INTERPRETER_ENTRYPOINT)
+        )
+        if entry_point == PEX_INTERPRETER_ENTRYPOINT and len(sys.argv) > 1:
+            args = sys.argv[1:]
+            arg = args[0]
+            if arg == "-m":
+                if len(args) < 2:
+                    sys.stderr.write("Argument expected for the -m option\\n")
+                    sys.exit(2)
+                entry_point = module = args[1]
+                sys.argv = args[1:]
+                # Fall through to entry_point handling below.
+            else:
+                filename = arg
+                sys.argv = args
+                if arg == "-c":
+                    if len(args) < 2:
+                        sys.stderr.write("Argument expected for the -c option\\n")
+                        sys.exit(2)
+                    filename = "-c <cmd>"
+                    content = args[1]
+                    sys.argv = ["-c"] + args[2:]
+                elif arg == "-":
+                    content = sys.stdin.read()
+                else:
+                    with open(arg) as fp:
+                        content = fp.read()
+
+                ast = compile(content, filename, "exec", flags=0, dont_inherit=1)
+                globals_map = globals().copy()
+                globals_map["__name__"] = "__main__"
+                globals_map["__file__"] = filename
+                locals_map = globals_map
+                {exec_ast}
+                sys.exit(0)
+
+        module_name, _, function = entry_point.partition(":")
+        if not function:
+            import runpy
+            runpy.run_module(module_name, run_name="__main__")
+        else:
+            import importlib
+            module = importlib.import_module(module_name)
+            # N.B.: Functions may be hung off top-level objects in the module namespace,
+            # e.g.: Class.method; so we drill down through any attributes to the final function
+            # object.
+            namespace, func = module, None
+            for attr in function.split("."):
+                func = namespace = getattr(namespace, attr)
+            func()
+        """.format(
+            venv_python=venv_python,
+            venv_bin_dir=venv_bin_dir,
+            venv_dir=venv_dir,
+            bin_path=bin_path,
+            entry_point=pex_info.entry_point,
+            exec_ast=(
+                "exec ast in globals_map, locals_map"
+                if venv.interpreter.version[0] == 2
+                else "exec(ast, globals_map, locals_map)"
+            ),
```

Comment:
Maybe not worth it yet, but if all of the templated variables are strings, you might consider extracting this as a "real" Python file to allow for mypy/linting, and then loading the file as a resource to template it using a slightly heavier template syntax (something like mustache, but cheaper hopefully).

Reply:
Perhaps worth it down the line as with the venv `__main__.py` and the `PythonInterpreter` identification code, etc...